<launch>

    <!-- Spawn at course start -->
   <arg name="x"     value="-12.79"/>
   <arg name="y"     value="18.79"/>
   <arg name="z"     value="0.25"/>
   <arg name="roll"  value="0"/>
   <arg name="pitch" value="0"/>
   <arg name="yaw"   value="0"/>

    <!-- Parse URDF with xacro -->
    <param name="robot_description" command="xacro '$(find description)/urdf/caffeine.urdf.xacro'"/>

    <!-- Spawn Caffeine in Gazebo -->
    <node name="spawn_caffeine" pkg="gazebo_ros" type="spawn_model"
        args="-param robot_description
            -urdf -model caffeine
            -x $(arg x) -y $(arg y) -z $(arg z)
            -R $(arg roll) -P $(arg pitch) -Y $(arg yaw)" />
    
    <!-- Set up TF Tree -->
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
        <param name="rate" value="50"/>
    </node>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    <!-- Twist Multiplexer -->
    <include file="$(find twist_mux)/launch/twist_mux.launch">
        <arg name="cmd_vel_out" value="cmd_vel"/>
    </include>

    <!-- Emulate ZED stereo camera image processing -->
    <include file="$(find description)/launch/zed_emulation.launch">
        <arg name="camera_ns" value="zed/zed_node"/>
    </include>

    <!-- Compute Odometry from Sensor Fusion -->
    <include file="$(find odom)/launch/odom.launch"/>

</launch>
